snowsql/resource_exec.go

```diff
@@ -136,7 +169,17 @@ func resourceExecRead(ctx context.Context, d *schema.ResourceData, m interface{}
 func resourceExecUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 
```

Comment:
Let's also get the latest value of `name` just like we do in the create function. Then compare the new name with state. If we have a change we should update the Id and replace the object in state. For example:

```
resource "snowsql_exec" "dcl" {
  name = "my-name"

  create {
    statements = "CREATE USER IF NOT EXISTS user1;"
  }

  delete {
    statements = "DROP USER IF EXISTS user1;"
  }
}
```

and 

```
resource "snowsql_exec" "dcl" {
  name = "my-name-changed-should-cause-replacement"

  create {
    statements = "CREATE USER IF NOT EXISTS user1;"
  }

  delete {
    statements = "DROP USER IF EXISTS user1;"
  }
}
```

Reply:
I didn't test that, but doesn't name change force a replacement anyways as it is `ForceNew: true`?
https://github.com/aidanmelen/terraform-provider-snowsql/blob/main/snowsql/resource_exec.go#L67-L70